### PR TITLE
Add a token generator to slack client.

### LINKED ISSUE #8843: prow: Dynamically reload token in slack client.
Related to #8579

### PR DESCRIPTION
**Pull request by droslean**
closes #8843

### CODE REVIEW

File: prow/cmd/hook/main.go [hunk 1]
```diff
@@ -156,9 +154,9 @@ func main() {
 	}
 
 	var slackClient *slack.Client
-	if !o.dryRun && teamToken != "" {
+	if !o.dryRun && string(secretAgent.GetSecret(o.slackTokenFile)) != "" {
```
Comment (kargakis) [inline prow/cmd/hook/main.go]:
```go
if !o.dryRun && o.slackTokenFile != "" {
```
is a bit cleaner

Comment (stevekuznetsov) [reply prow/cmd/hook/main.go]: Asking for a token that is not stored from the agent sounds like it should error out but as implemented this seems OK?

**Comment by stevekuznetsov** [comment]
/lgtm
/hold
@kargakis your call

**Comment by kargakis** [comment]
this is fine too

/hold cancel

**Pull request merged**